File: Kids With the Greatest Number of Candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        
        vector<bool>v;
        int n = candies.size() , maxCandies = 0;
        for(int i = 0 ; i < n ; i++)
            if(candies[i] > maxCandies)
               maxCandies = candies[i];
        
        for(int i=0;i<n;i++){
            if((candies[i]+extraCandies)>=maxCandies){
                v.push_back(true);
            }else{
                v.push_back(false);
            }
        }
        return v;
    }
};

File: Kids With the Greatest Number of Candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        boolean res=false;
        int n=candies.length;
        ArrayList <Boolean> result=new ArrayList<Boolean>();
        for(int i=0;i<candies.length;i++)
        {
              int sum=0;
            sum=candies[i]+extraCandies;
            res=max(sum,candies);
           result.add(res);
        }
        return result;
    }
    
    boolean max(int sum,int[] candies)
    {
        for(int i=0;i<candies.length;i++)
        {
            if(candies[i]<=sum)
                continue;
            else
                return false;
                }
        return true;
    }
}
